File: 792. Binary Search/Solution.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int i = 0;
        int j = nums.size()-1;
        int k = nums.size() / 2;

        while (i <= j){
            if (nums[k] == target){
                return k;
            }
            if (nums[k] < target)
                i = k+1;
            else 
                j = k-1;
            k = (i+j)/2;
        }

        return -1;
    }
};

File: 792. Binary Search/Solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums)-1

        while i<=j:
            k = (i+j)//2
            if nums[k] < target:
                i = k+1
            elif nums[k] > target:
                j = k-1
            elif nums[k] == target:
                return k
        return -1